packages/browser-sdk/src/feedback/feedback.ts

```diff
@@ -0,0 +1,456 @@
+    private httpClient: HttpClient,
+    private feedbackPromptHandler: FeedbackPromptHandler = createDefaultFeedbackPromptHandler(),
+    private userId: string,
+    private position: FeedbackPosition = DEFAULT_POSITION,
+    private feedbackTranslations: Partial<FeedbackTranslations> = {},
+  ) {}
+
+  /**
+   * Start receiving Live Satisfaction feedback prompts.
+   */
+  async init() {
+    if (this.initialized) {
+      this.logger.error("feedback prompting already initialized");
+      return;
+    }
+
+    const existingAuth = getAuthToken(this.userId);
+    let channel = existingAuth?.channel;
+
+    // while initializing, consider the channel active
+    // liveSatisfactionActive = true;
+    try {
+      if (!channel) {
+        const res = await this.httpClient.post({
+          path: `/feedback/prompting-init`,
+          body: {
+            userId: this.userId,
+          },
+        });
+
+        this.logger.debug(`feedback prompting status sent`, res);
+        const body: { success: boolean; channel?: string } = await res.json();
+        if (!body.success || !body.channel) {
+          this.logger.debug(`feedback prompting not enabled: ${body}`);
+          return;
+        }
+
+        channel = body.channel;
+      }
+
+      this.logger.debug(`feedback prompting enabled`, channel);
+
+      this.sseChannel = openAblySSEChannel({
+        userId: this.userId,
+        channel,
+        httpClient: this.httpClient,
+        callback: (message) =>
+          this.handleFeedbackPromptRequest(this.userId, message),
+        logger: this.logger,
+        sseHost: this.sseHost,
+      });
+
+      this.logger.debug(`feedback prompting connection established`);
+    } catch (e) {
+      this.logger.error(`error initializing feedback prompting`, e);
+      return;
+    }
+    return;
+    // return channel;
+  }
+
+  handleFeedbackPromptRequest(userId: string, message: any) {
+    const parsed = parsePromptMessage(message);
+    if (!parsed) {
+      this.logger.error(`invalid feedback prompt message received`, message);
+    } else {
+      if (
+        !processPromptMessage(userId, parsed, async (u, m, cb) => {
+          await this.feedbackPromptEvent({
+            promptId: parsed.promptId,
+            featureId: parsed.featureId,
+            promptedQuestion: parsed.question,
+            event: "received",
+            userId,
+          });
+          await this.triggerFeedbackPrompt(u, m, cb);
+        })
+      ) {
+        this.logger.info(
+          `feedback prompt not shown, it was either expired or already processed`,
+          message,
+        );
+      }
+    }
+  }
+
+  stop() {
+    if (this.sseChannel) {
+      this.sseChannel.close();
+      this.sseChannel = null;
+    }
+  }
+
+  async triggerFeedbackPrompt(
+    userId: string,
+    message: FeedbackPrompt,
+    completionHandler: FeedbackPromptCompletionHandler,
+  ) {
+    let feedbackId: string | undefined = undefined;
+
+    // TODO: still needed?
```

Comment:
is needed if we allow switching users after the SDK has been initialized. Otherwise, not really needed.